Classes/Backend/FormDataProvider/ModifyRootUidOfTreeSelectElements.php

```diff
@@ -51,7 +45,7 @@ public function addData(array $result): array
 
                 // check if a FlexForm was rendered
                 && $result['tableName'] === 'tt_content'
-                && GeneralUtility::isFirstPartOfStr($result['flexParentDatabaseRow']['list_type'], 'events2')
+                && \str_starts_with($result['flexParentDatabaseRow']['list_type'], 'events2')
             ) {
                 $typo3Version = GeneralUtility::makeInstance(Typo3Version::class);
```

Comment:
is it possible to move this outside the for loop?
Also the array frequently accessed $result['processedTca']['columns'] can be assigned to a variable to check inside the condition as it is used 3 or more times inside the condition.

Reply:
I have inserted it as constructor argument